python/ray/util/spark/databricks_hook.py

```
@@ -54,21 +58,103 @@ def display_databricks_driver_proxy_url(spark_context, port, title):
+    """
+    return get_dbutils().entry_point
+
+
+_DATABRICKS_DEFAULT_TMP_DIR = "/local_disk0/tmp"
+
+
 class DefaultDatabricksRayOnSparkStartHook(RayOnSparkStartHook):
     def get_default_temp_dir(self):
-        return "/local_disk0/tmp"
+        return _DATABRICKS_DEFAULT_TMP_DIR
 
     def on_ray_dashboard_created(self, port):
         display_databricks_driver_proxy_url(
             get_spark_session().sparkContext, port, "Ray Cluster Dashboard"
         )
 
-    def on_spark_background_job_created(self, job_group_id):
+    def on_cluster_created(self, ray_cluster_handler):
+        db_api_entry = _get_db_api_entry()
         try:
-            get_dbutils().entry_point.registerBackgroundSparkJobGroup(job_group_id)
+            db_api_entry.registerBackgroundSparkJobGroup(
+                ray_cluster_handler.spark_job_group_id
+            )
+        except Exception:
+            _logger.warning(
+                "Registering ray cluster spark job as background job failed. "
+                "You need to manually call `ray.util.spark.shutdown_ray_cluster()` "
+                "before detaching your databricks notebook."
+            )
+
+        auto_shutdown_timeout_minutes = float(
+            os.environ.get(DATABRICKS_RAY_ON_SPARK_AUTOSHUTDOWN_TIMEOUT_MINUTES, "30")
+        )
+        if auto_shutdown_timeout_minutes == 0:
+            _logger.info(
+                "The Ray cluster will keep running until you manually detach the "
+                "databricks notebook or call "
+                "`ray.util.spark.shutdown_ray_cluster()`."
+            )
+        elif auto_shutdown_timeout_minutes < 0:
+            _logger.warning(
+                "You must set "
+                f"'{DATABRICKS_RAY_ON_SPARK_AUTOSHUTDOWN_TIMEOUT_MINUTES}' "
+                "to a value >= 0. The negative value you set is ignored and the "
+                "default value 30 will be used."
+            )
+            auto_shutdown_timeout_minutes = 30
```

Can we throw an exception instead of defaulting to 30?